docker/scripts/dump-env.sh

```
@@ -0,0 +1,137 @@
+env | while IFS='=' read -r name value; do
+  # Skip empty names (shouldn't happen, but be safe)
+  [ -z "$name" ] && continue
+
+  # Check if this variable should be excluded or is not known
+  if ! should_exclude_var "$name" && is_known_var "$name"; then
+    # Handle values that may contain special characters or newlines
+    # We need to properly escape the value for the .env file
+
+    # If value contains newline, quotes, or special chars, wrap in quotes
+    case "$value" in
+    *$'\n'* | *\"* | *\'* | *\$* | *\`* | *\\* | *" "* | *"="*)
```

Shellcheck (0.11.0)

[warning] In POSIX sh, $'..' is undefined.

(SC3003)